src/service/chat.js

```diff
@@ -203,6 +203,58 @@ const moveChatToAnon = async ({
   }
 };
 
+const initChatFromPost = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await api.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
+};
+
+const initChatFromPostAnon = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await anonymousApi.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChatFromPostAnon` function is almost identical to the `initChatFromPost` function. This is a violation of the DRY (Don't Repeat Yourself) principle. Consider refactoring these functions to avoid code duplication. You could create a single function that accepts an additional parameter to determine whether to use `api.post` or `anonymousApi.post`.
```javascript
const initChatFromPost = async ({source, id}, isAnonymous = false) => {
  let payload = {source};
  switch (source) {
    case 'post':
      payload = {...payload, postId: id};
      break;
    case 'comment':
      payload = {...payload, commentId: id};
      break;
    default:
      throw new Error(`Invalid source: ${source}`);
  }

  try {
    const apiInstance = isAnonymous ? anonymousApi : api;
    const response = await apiInstance.post('/chat/init-chat-from-post', payload);
    if (response?.status === 200) {
      return Promise.resolve(response.data);
    }

    return Promise.reject(response.data);
  } catch (e) {
    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
    return Promise.reject(e);
  }
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->